node/src/neighborhood/mod.rs

```diff
@@ -562,7 +560,7 @@ impl Neighborhood {
                     self.user_exit_preferences.locations_opt.clone()
                 {
                     for exit_location in &exit_locations_by_priority {
-                        self.enrich_exit_countries_returns_missing(&exit_location.country_codes);
+                        self.enrich_exit_countries_and_return_missing(&exit_location.country_codes);
```

Comment:
How about `synchronize` exit countries, since you could be both adding and subtracting?